src/core/serialization.rs

```diff
@@ -80,6 +80,7 @@ impl Serializable for MasterSecretKey {
 
     fn length(&self) -> usize {
         let mut length = 3 * R25519PrivateKey::LENGTH
+            + SYM_KEY_LENGTH
```

Comment:
This is true only if there is a KMAC key:
```suggestion
            + self.kmac_key.as_ref().map_or_else(|| 0, |key| key.len())
```